shell/components/form/NodeScheduling.vue

```diff
@@ -127,7 +127,33 @@ export default {
       }
     },
     isEmpty
-  }
+  },
+
+  watch: {
+    'value.nodeSelector': {
+      handler(nodeSelector) {
+        if (this.isHarvester && nodeSelector?.[HOSTNAME]) {
+          this.selectNode = 'nodeSelector';
+          const nodeName = nodeSelector[HOSTNAME];
+
+          this.nodeName = nodeName;
+
+          const array = this.nodes.map(n => n.value);
+
+          if (nodeName && !array.includes(nodeName)) {
+            this.$store.dispatch('growl/error', {
+              title:   this.$store.getters['i18n/t']('harvester.vmTemplate.tips.notExistNode.title', { name: nodeName }),
+              message: this.$store.getters['i18n/t']('harvester.vmTemplate.tips.notExistNode.message')
+            }, { root: true });
+          }
+
+          delete this.value.nodeSelector;
+          this.$set(this, 'nodeName', '');
```

Comment:
I wonder if we could move this section to mixins/harvester-vm, which would affect the rancher's logic.

Reply:
I am afraid not, need to set nodeName and selectNode which belongs to the component to empty.